Flow.Launcher/ViewModel/MainViewModel.cs

```diff
@@ -701,11 +711,31 @@ private ResultsViewModel SelectedResults
                     // result from the one that was selected before going into the context menu to the first result.
                     // The code below correctly restores QueryText and puts the text caret at the end without
                     // running the query again when returning from the context menu.
-                    if (isReturningFromContextMenu)
+                    if (isReturningFromContextMenu && _lastSelectedResultItem != null)
                     {
-                        _queryText = _queryTextBeforeLeaveResults;
-                        OnPropertyChanged(nameof(QueryText));
-                        QueryTextCursorMovedToEnd = true;
+                        try
+                        {
+                            _blockQueryExecution = true;
+                            // Set querytext without running the query for keep index. if not use this, index will be 0.
+                            _queryText = _queryTextBeforeLeaveResults;
+                            OnPropertyChanged(nameof(QueryText));
+                            QueryTextCursorMovedToEnd = true;
+                        }
+                        finally
+                        {
+                            _blockQueryExecution = false;
+                        }
+                        // restore selected item
+                        if (Results.Results.Contains(_lastSelectedResultItem))
```

Comment:
Would this be slow for large results? And I am curious when will Results not contains the last selected result item? Why not just use _lastSelectedIndex

Reply:
OK. Modified it to work using only the index.